File: .claude/hooks/post-compact-save-summary.mjs
#!/usr/bin/env node
// @hookstack post-compact-save-summary
// Journalise le résumé de compaction dans .claude/compaction-log.md (PostCompact)
import { appendFileSync, mkdirSync, readFileSync } from "node:fs";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";

export function run(
	input,
	{
		append = appendFileSync,
		mkdir = mkdirSync,
		projectDir = process.env.CLAUDE_PROJECT_DIR ?? ".",
		now = () => new Date().toISOString(),
	} = {},
) {
	const summary = input.compact_summary ?? "";
	if (!summary.trim()) return null;

	const logPath = join(projectDir, ".claude", "compaction-log.md");
	try {
		mkdir(dirname(logPath), { recursive: true });
	} catch {
		/* exists */
	}

	const entry = `\n## ${now()} (${input.trigger ?? "auto"})\n${summary}\n`;
	append(logPath, entry);
	return entry;
}

/* v8 ignore next 4 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
	const input = JSON.parse(readFileSync(0, "utf8"));
	run(input);
}

File: .claude/hooks/post-compact-save-summary.py
#!/usr/bin/env python3
# @hookstack post-compact-save-summary
"""Logs the compaction summary into .claude/compaction-log.md (PostCompact)."""
import json
import os
import sys
from datetime import datetime, timezone


def _now():
    return datetime.now(timezone.utc).isoformat()


def _mkdir(path):
    os.makedirs(path, exist_ok=True)


def _append(path, content):
    with open(path, "a", encoding="utf8") as f:
        f.write(content)


def run(input_data, *, append=None, mkdir=None, project_dir=None, now=None):
    if append is None:
        append = _append
    if mkdir is None:
        mkdir = _mkdir
    if project_dir is None:
        project_dir = os.environ.get("CLAUDE_PROJECT_DIR", ".")
    if now is None:
        now = _now

    summary = input_data.get("compact_summary") or ""
    if not summary.strip():
        return None

    log_path = os.path.join(project_dir, ".claude", "compaction-log.md")
    try:
        mkdir(os.path.dirname(log_path))
    except Exception:
        pass

    entry = f"\n## {now()} ({input_data.get('trigger') or 'auto'})\n{summary}\n"
    append(log_path, entry)
    return entry


if __name__ == "__main__":
    data = json.load(sys.stdin)
    run(data)
